### PR TITLE
Remove ufc geometry and get geometry data from Basix

### PR DESCRIPTION
**Pull request by mscroggs**

### CODE REVIEW

File: ffcx/codegeneration/geometry.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+
+def reference_facet_jacobian(L, tablename, cellname):
+    celltype = getattr(basix.CellType, cellname)
+    topology = basix.topology(celltype)
+    geometry = basix.geometry(celltype)
+
+    tdim = len(topology) - 1
+
+    if tdim < 2:
```
Comment (michalhabera) [inline ffcx/codegeneration/geometry.py:54]: This check could throw for `if tdim not in [2, 3]` and avoid the assert below.

File: ffcx/codegeneration/integrals.py [hunk 2]
```diff
@@ -186,6 +187,10 @@ def generate(self):
         # blocks
         parts += self.generate_element_tables()
 
+        # Generate the tables of basis function values and preintegrated
```
Comment (michalhabera) [inline ffcx/codegeneration/integrals.py:190]: This comment is wrong.

File: ffcx/codegeneration/geometry.py [hunk 3]
```diff
@@ -0,0 +1,170 @@
+import basix
```
Comment (michalhabera) [inline ffcx/codegeneration/geometry.py:1]: Add the header copyright so we're consistent.

File: ffcx/codegeneration/geometry.py [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+            edge1 = geometry[facet[2]] - geometry[facet[0]]
+            facet_jacobian.append(list(zip(edge0, edge1)))
+
+    out = numpy.array(facet_jacobian)
+    return L.ArrayDecl("static const double", f"{cellname}_{tablename}", out.shape, out)
+
+
+def reference_cell_volume(L, tablename, cellname):
+    cell_volume = {
+        "interval": 1.0, "triangle": 0.5, "tetrahedron": 1 / 6,
```
Comment (michalhabera) [inline ffcx/codegeneration/geometry.py:76]: Isn't this basix dependent?

Comment (michalhabera) [reply ffcx/codegeneration/geometry.py:76]: Ok, I see basix issue.

**Review by chrisrichardson — APPROVED** [review]

**Review by garth-wells — APPROVED** [review]

**Pull request merged**